### ISSUE TITLE
Extract the GCL-related code from stackdriver into a separate package

### ISSUE THREAD

**Issue by BorzdeG** [dusted-go/logging#10]
The stackdriver package contains code that can be used not only for the GCL stack - for example, tracing. I suggest dividing this into logical blocks

**Comment by dustinmoris** [comment]
I think this is a great suggestion, I might pick this up over the weekend or if you want you can do a head start and create a branch with the refactorings as you see fit and I can help and review 🙏

**Comment by BorzdeG** [comment]
Yes, I was going to start creating PR for all my issues in the coming days...

**Comment by dustinmoris** [comment]
Should be addressed via https://github.com/dusted-go/logging/releases/tag/v2.0.0-rc-01
